test/Api.Tests/ControllerTests/UserControllerTest.cs

```diff
@@ -0,0 +1,135 @@
+using PaderbornUniversity.SILab.Hip.CmsApi.Models.User;
+using Xunit;
+
+namespace PaderbornUniversity.SILab.Hip.CmsApi.Tests.ControllerTests
+{
+    public class UserControllerTest
+    {
+        private ControllerTester<UserController> _tester;
+        public UserFormModel UserFormModel { get; set; }
+
+        public UserControllerTest()
+        {
+            _tester = new ControllerTester<UserController>();
+            UserFormModel = new UserFormModel
+            {
+                FirstName = "First Name",
+                LastName = "Last Name"
+            };
+        }
+
+        #region Get User
+
+        /// <summary>
+        /// Should return 200 when trying to get user details
+        /// </summary>
+        [Fact]
+        public void GetUserTest200()
+        {
+            _tester.TestController()
+                .Calling(c => c.Get(_tester.Admin.Email))
+                .ShouldReturn()
+                .Ok()
+                .WithStatusCode(200)
+                .WithModelOfType<UserResult>()
+                .Passing(m => m.Email == _tester.Admin.Email);                
+        }
+
+        /// <summary>
+        /// Should return 200 when trying to get user details
+        /// </summary>
+        [Fact]
+        public void GetUserTest200ForStudent()
+        {
+            _tester.TestController()
+                .Calling(c => c.Get(_tester.Student.Email))
+                .ShouldReturn()
+                .Ok()
+                .WithStatusCode(200)
+                .WithModelOfType<UserResult>()
+                .Passing(m => m.Email == _tester.Student.Email);
+        }
+
+        /// <summary>
+        /// Should return 404 when trying to get user details
+        /// </summary>
+        [Fact]
+        public void GetUserTest404()
+        {
+            _tester.TestController()
+                .Calling(c => c.Get("hello@hipapp.de"))
+                .ShouldReturn()
+                .NotFound();
+        }
+
+        #endregion
+
+        #region PUT user tests
+
+        /// <summary>
+        /// Should return code 200 if user is updated
+        /// </summary>
+        [Fact]
+        public void PutUserTest200()
+        {
+            _tester.TestController()
+                .Calling(c => c.Put(_tester.Admin.Email, UserFormModel))
+                .ShouldHave()
+                .DbContext(db => db.WithSet<User>(user =>
+                    user.Find(_tester.Admin.Id).FirstName.Equals(UserFormModel.FirstName) &&
+                    user.Find(_tester.Admin.Id).FirstName.Equals(UserFormModel.FirstName)
+                ))
+                .AndAlso()
+                .ShouldReturn()
+                .Ok()
+                .WithStatusCode(200);
+        }
+
+        /// <summary>
+        /// Should return bad request when trying to update user with invalid role
+        /// </summary>
+        [Fact]
+        public void PutUserTest400()
+        {
+            UserFormModel.Role = "InvalidRole";
```

Comment:
create a new local variable for the `UserFormModel` with the invalid role. The reason is that this assignment changes the `Role` of an object that other tests have access to (and those would possibly fail if the order of tests change)